htdocs/install/class/pathcontroller.php

```
@@ -233,22 +233,98 @@ public function sanitizePath($path)
+            }
+            $nameIndex = $this->nextSignificantTokenIndex($tokens, $openParenIndex + 1);
+            if (null === $nameIndex || !is_array($tokens[$nameIndex]) || $tokens[$nameIndex][0] !== T_CONSTANT_ENCAPSED_STRING) {
+                continue;
+            }
+            if ('XOOPS_VERSION' === trim($tokens[$nameIndex][1], "\"'")) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

PHPMD (2.15.0)

[warning] The method hasXoopsVersionDefinition() has a Cyclomatic Complexity of 11. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)